corehq/apps/translations/generators.py

```diff
@@ -47,18 +133,20 @@ def __init__(self, domain, app_id, version, key_lang, source_lang, lang_prefix,
                             "that would result in empty msgstr and no display for other lang")
         self.domain = domain
         self.app_id = app_id
+        self.app = get_current_app(domain, app_id)
         self.key_lang = key_lang
         self.source_lang = source_lang
         self.lang_prefix = lang_prefix
         self.exclude_if_default = exclude_if_default
-        self.translations = OrderedDict()
-        self.version = version
+        self.version = self.app.version if version is None else version
         self.use_version_postfix = use_version_postfix
+        self.checker = EligibleForTransifexChecker(self.app)
+
         self.headers = dict()  # headers for each sheet name
         self.sheet_name_to_module_or_form_type_and_id = dict()
         self.slug_to_name = defaultdict(dict)
         self.slug_to_name[MODULES_AND_FORMS_SHEET_NAME] = {'en': MODULES_AND_FORMS_SHEET_NAME}
-        self._build_translations()
+        self.translations = self._build_translations()
 
     @cached_property
     def build_id(self):
```

Comment:
@emord does this get used anywhere?

Reply:
I think here: https://github.com/dimagi/commcare-hq/blob/4119e3b6b390ae3eca91fa7d89564296596e5cee/corehq/apps/translations/integrations/transifex/parser.py#L35 ?

Reply:
:+1: Thanks. That looks right.